nf_core/modules/pipeline_modules.py

```diff
@@ -198,7 +198,8 @@ def list_modules(self, print_json=False):
     def install(self, module=None):
 
         # Check whether pipelines is valid
-        self.has_valid_pipeline()
+        if self.has_valid_pipeline():
+            self.has_modules_file()
```

Comment:
Maybe it makes sense to call `self.has_modules_file` directly within `self.has_valid_pipeline`

Reply:
`nf-core modules remove` also calls `self.has_valid_pipeline` but I figure the `modules.json` file should be part of every valid DSL2 pipeline from now on.

Reply:
Yes, and in a way, we're checking whether it's a valid pipeline and so if it doesn't have a `modules.json` file, it won't be valid in the future.
And those module commands shouldn't be used for DSL1 pipelines.